Fix intersection for symbolics

madt2709 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->
Fixes #25222


#### Brief description of what is fixed or changed
`Plane.intersection` now supports symbolic case by returning `Piecewise` object. It effectively replicates the old logic but turns the conditions into equations to satisfy. 

#### Other comments
Currently, the plane intersecting another plane case won't work. This is due to https://github.com/sympy/sympy/issues/25228. This function is called when assessing `if self.equals(o)`. 

I will make a quick fix for the above and then we should re run the tests on this PR.

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* geometry
  * Added symbolic case to `Plane.intersection`. Previously, the results returned were wrong in the symbolic case.
<!-- END RELEASE NOTES -->

madt2709 committed:
Support symbolic case

madt2709 committed:
Add tests

1e9abhi1e10 commented on sympy/geometry/plane.py:
```diff
@@ -384,11 +386,15 @@ def intersection(self, o):
                 return [o]
+            # Check for symbolic case.
+            elif any(isinstance(i, Symbol) for i in o) or any(i for i in symbolic_self):
+                return [Piecewise((o,Eq((o-self.p1).dot(Point3D(self.normal_vector)),0)),(set(),True))]
```
Should be a space after a comma

madt2709 replied:
Good spot, will update

1e9abhi1e10 commented:
Looks like it is your first PR, you need to add your name and email in the `.mailmap` file.
Check [instructions](https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-process.html#add-your-name-and-email-address-to-the-mailmap-file).

madt2709 commented:
@1e9abhi1e10 I would like for https://github.com/sympy/sympy/pull/25231 to be merged first. I added my author name on that PR instead. Once https://github.com/sympy/sympy/pull/25231 is merged than we can merge this PR.

madt2709 committed:
Add space after commas

madt2709 committed:
Rename test variables to avoid conflicts

madt2709 committed:
author: add Theo Dias to .mailmap

madt2709 commented:
@1e9abhi1e10 we've opted only to raise an error message in case of a clash of variables. Therefore, I have updated the variable names in the tests and added my author name here. Please approve if you think this can go in 🙏
